Standalone: bgfx init segfaults on wayland #2157

superhac opened the pull request:
This is a fix related to [2157](https://github.com/vpinball/vpinball/issues/2157).  Currently this stops the segfault but the pinmame window gets rendered behind the b2s.  In other words, it doesn't seem to be respecting the cords set for where the pinmame window should reside.   Not sure what the problem is.

francisdb commented:
Wayland has no window positioning protocol so windows will appear wherever wayland decides to put them.
There's a limited protocol merge request that won't help us.
https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/264

For desktop use this is of course a great improvement no longer requiring X11 with env vars

francisdb commented on platforms/linux-x64/external.sh:
```diff
@@ -165,6 +165,7 @@ if [ "${BGFX_EXPECTED_SHA}" != "${BGFX_FOUND_SHA}" ]; then
       -DBGFX_BUILD_EXAMPLES=OFF \
       -DBGFX_CONFIG_MULTITHREADED=ON \
       -DBGFX_CONFIG_MAX_FRAME_BUFFERS=256 \
+      -DBGFX_WITH_WAYLAND=ON \
```
I don't think this is required as this is set automatically on linux is my understanding of below linked line is correct.

see https://github.com/bkaradzic/bgfx.cmake/blob/36a0d1d8d040113a954f99b1456121072dc77348/CMakeLists.txt#L46

francisdb reviewed (approved):
I can confirm this fixes the crash on Wayland. There are other issues like window sizes not being respected but let's keep those for later.

jsm174 reviewed (commented):
I don’t think you need to create a whole new structure. Can’t you just set the type directly?

superhac commented:
> I don’t think you need to create a whole new structure. Can’t you just set the type directly?

Your right.  I can access it through bgfx::Init struct.  Fixed.   I also removed the compile option for wayland in the bgfx section of the `external.sh`.

superhac commented:
So at least on mine in cab mode, the table and BS2 screen are on the right monitor.  But the pinmame window is behind the B2S for some reason.
